### ISSUE TITLE
Error with backup command

### ISSUE THREAD

**Issue by cognotte** [teslamate-org/teslamate#3518]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### What happened?

Hello,

when I want to launch a backup I use the following command : 

`docker compose exec -T database pg_dump -U teslamate teslamate > /home/myuser/teslamate.bck`

I get this error message in return: 

`service "teslamate" is not running`

the script is installed via docker-compose.

thanks in advance

### Expected Behavior

backup completed

### Steps To Reproduce

docker compose exec -T database pg_dump -U teslamate teslamate > /home/myuser/teslamate.bck

### Relevant log output

```shell
.
```

### Type of installation

Docker

### Version

v1.28.2

**Comment by cwanja** [comment]
Going to ask the obvious, is `teslamate` running? Is TeslaMate the name of the service in your YML file?

**Comment by cognotte** [comment]
Good evening

yes teslamate is working.

and this is indeed the name of the container in the yml.

**Comment by cwanja** [comment]
What is the output of `docker compose ps`? Ironically, I do not _think_ **TeslaMate** web service has to be actually running - just the **database** service. So that is an odd error to throw. 

With the default YML from docs.teslamate.org and the exact same command, I have zero issues. Just updated TeslaMate to 1.28.2

**Comment by cognotte** [comment]
my stack

`version: "3"

services:
  teslamate:
    image: teslamate/teslamate:latest
    restart: unless-stopped
    environment:
      - ENCRYPTION_KEY=************
      - DATABASE_USER=teslamate
      - DATABASE_PASS=**********
      - DATABASE_NAME=teslamate
      - DATABASE_HOST=database
      - MQTT_HOST=mosquitto
    ports:
      - 4000:4000
    volumes:
      - ./import:/opt/app/import
    cap_drop:
      - all


  database:
    image: postgres:14
    restart: unless-stopped
    environment:
      - POSTGRES_USER=teslamate
      - POSTGRES_PASSWORD=**********
      - POSTGRES_DB=teslamate
    ports:
      - 5432:5432
    volumes:
      - teslamate-db:/var/lib/postgresql/data
      
      

  grafana:
    image: teslamate/grafana:latest
    restart: unless-stopped
    environment:
      - DATABASE_USER=teslamate
      - DATABASE_PASS=***********
      - DATABASE_NAME=teslamate
      - DATABASE_HOST=database
    ports:
      - 3000:3000
    volumes:
      - teslamate-grafana-data:/var/lib/grafana

  mosquitto:
    image: eclipse-mosquitto:2
    restart: unless-stopped
    command: mosquitto -c /mosquitto-no-auth.conf
    ports:
      - 1883:1883
    volumes:
      - mosquitto-conf:/mosquitto/config
      - mosquitto-data:/mosquitto/data



volumes:
  teslamate-db:
  teslamate-grafana-data:
  mosquitto-conf:
  mosquitto-data:

networks:
  teslamate_default:`

**Comment by cwanja** [comment]
What is the output of `docker compose ps` and or `docker ps`?

**Comment by cognotte** [comment]
![chrome_uyDtzfFHn7](https://github.com/teslamate-org/teslamate/assets/6647413/4e170d37-6693-4461-93f5-6eff3774f84a)

**Comment by cognotte** [comment]
On the other hand, docker compose ps does not give anything.

**Comment by miga97** [comment]
if you are using the stack and not the compose you should use the command without compose

docker exec -t [container name] pg_dump -U teslamate teslamate > /home/myuser/teslamate.bck

works for me

but I haven't been able to do a restore yet without using compose

**Comment by cognotte** [comment]
> si vous utilisez la pile et non la composition, vous devez utiliser la commande sans composer
> 
> docker exec -t [nom du conteneur] pg_dump -U teslamate teslamate > /home/myuser/teslamate.bck
> 
> travaille pour moi
> 
> mais je n'ai pas encore pu effectuer de restauration sans utiliser Compose

Good evening,

Indeed, I got an error when trying to restore. On the other hand, thank you I managed to make my backup.


root@unraidmox:~# docker exec -T teslamate_database_1 psql -U teslamate -d teslamate < mnt/disk3/teslamate.bck
unknown shorthand flag: 'T' in -T
See 'docker exec --help'.

**Comment by miga97** [comment]
> Good evening,
> 
> Indeed, I got an error when trying to restore. On the other hand, thank you I managed to make my backup.
> 
> root@unraidmox:~# docker exec -T teslamate_database_1 psql -U teslamate -d teslamate < mnt/disk3/teslamate.bck unknown shorthand flag: 'T' in -T See 'docker exec --help'.

there is no flag -T in the "no compose" exec command.
In theory you should use one of the following flags: -t -i -it  or no flag instead of -T
but I haven't found a way to make it work

there are some things that I didn't understand, but the backup should be lines of text that contain queries that re-populate the database, which is why to do the restore you are asked to create the functions in the database with that long command, like explained [here](https://docs.teslamate.org/docs/maintenance/backup_restore/#restore)

I found other commands that create backups with different formats ( [source](https://www.postgresguide.com/utilities/backup-restore/) ), but I think that the reason why the backup is done like this is because it must be possible to pass the commands via string from outside the container

the solution I adopted was to connect to the container via bash and execute the commands directly inside the container, moving the files via a folder that I bind

**Comment by JakobLichterfeld** [comment]
Postgres DB Backup and restore only works with same postgres version...

**Comment by DrMichael** [comment]
> Postgres DB Backup and restore only works with same postgres version...

We probably should that add to the documentation in https://github.com/teslamate-org/teslamate/blob/master/website/docs/maintenance/backup_restore.md

**Comment by cwanja** [comment]
> > Postgres DB Backup and restore only works with same postgres version...
> 
> We probably should that add to the documentation in https://github.com/teslamate-org/teslamate/blob/master/website/docs/maintenance/backup_restore.md

Done #3560.

**Comment by JakobLichterfeld** [comment]
> > Postgres DB Backup and restore only works with same postgres version...
> 
> We probably should that add to the documentation in https://github.com/teslamate-org/teslamate/blob/master/website/docs/maintenance/backup_restore.md

No, as our process is fine. Do not use postgres backup feature. Use Teslamate backup feature.

**Comment by JakobLichterfeld** [comment]
> Done #3560.

and reverted :-)

**Comment by miga97** [comment]
> Postgres DB Backup and restore only works with same postgres version...
>
> Edit: to be clear: Postgres build in db dump can cause issues, thats why we use or own backup and restore commands.
@JakobLichterfeld 

I put the link to those methods just for information.

I used your guide to migrate the database and it worked, but I did it a little differently. I don't use docker compose, but instead I use docker stacks, as does the author of this issue. 

I was unable to make the restore command work because of the " -T " and the solution I used is the one I explained previously: I ran the commands from inside the container taking the files from a bind folder
